dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.xml

```diff
@@ -183,4 +183,14 @@
             and post_task_code = #{postTaskCode}
         </if>
     </select>
+    <select id="queryDistinctDefineCodeByTaskCode" resultType="org.apache.dolphinscheduler.dao.entity.ProcessTaskRelation">
+        select
+        process_definition_code
+        from t_ds_process_task_relation
+        WHERE pre_task_code = #{taskCode}
+        <if test="taskCode != 0">
+            or post_task_code = #{taskCode}
```

Comment:
I think it's better to split this sql instread of `or` query, which will dismatch index.